src/pages/RoomInvitePage.js

```diff
@@ -69,7 +69,10 @@ function RoomInvitePage(props) {
     const [userToInvite, setUserToInvite] = useState(null);
 
     // Any existing participants and Expensify emails should not be eligible for invitation
-    const excludedUsers = useMemo(() => [...lodashGet(props.report, 'participants', []), ...CONST.EXPENSIFY_EMAILS], [props.report]);
+    const excludedUsers = useMemo(
+        () => _.map([...lodashGet(props.report, 'participants', []), ...CONST.EXPENSIFY_EMAILS], (participant) => OptionsListUtils.addSMSDomainIfPhoneNumber(participant)),
+        [props.report],
+    );
```

Comment:
There is a minor change in my [original proposal](https://github.com/Expensify/App/issues/30190#issuecomment-1781520533)
- In the original, I will "add the @expensify.sms with the phone number when calling API "InviteToRoom".
- But when implementing the PR, I found that if we do like above, the unexpected personalProfile account will be returned by BE. So I will keep the current logic when calling API "InviteToRoom", and just add the "@expensify.sms" to the excludeUser